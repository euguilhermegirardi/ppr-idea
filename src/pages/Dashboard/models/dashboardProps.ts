export interface DashboardRepositoryProps {
    full_name: string;
    description: string;
    owner: {
        login: string;
        avatar_url: string;
    };
};