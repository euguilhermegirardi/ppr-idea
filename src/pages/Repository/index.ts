export { default } from './repository_container'