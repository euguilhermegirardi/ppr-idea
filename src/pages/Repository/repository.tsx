import React from 'react';
import { Link } from 'react-router-dom';
import { FiChevronLeft, FiChevronRight } from 'react-icons/fi';
import { Header, RepositoryInfo, Issues } from './styles';
import { RepositoryProps, Issue } from './models/repositoryProps';

const Repository = ({
    repository,
    issues,
}: {
    repository: RepositoryProps | null;
    issues: Issue[];
}) => {
    return (
        <>
            <Header>
                <Link to="/">
                    <FiChevronLeft size={16} />
                    Back
                </Link>
            </Header>

            {repository && (
                <RepositoryInfo>
                    <header>
                        <img
                            src={repository.owner.avatar_url}
                            alt={repository.owner.login}
                        />
                        <div>
                            <strong>{repository.full_name}</strong>
                            <p>{repository.description}</p>
                        </div>
                    </header>

                    <ul>
                        <li>
                            <strong>{repository.stargazers_count}</strong>
                            <span>Stars</span>
                        </li>
                        <li>
                            <strong>{repository.forks_count}</strong>
                            <span>Forks</span>
                        </li>
                        <li>
                            <strong>{repository.open_issues_count}</strong>
                            <span>Issues</span>
                        </li>
                    </ul>
                </RepositoryInfo>
            )}

            <Issues>
                {issues.map(issue => (
                    <a key={issue.id} href={issue.html_url}>
                        <div>
                            <strong>{issue.title}</strong>
                            <p>{issue.user.login}</p>
                        </div>

                        <FiChevronRight size={20} />
                    </a>
                ))}
            </Issues>
        </>
    );
};

export default Repository;