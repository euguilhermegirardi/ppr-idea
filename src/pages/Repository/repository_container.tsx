import { FC, useEffect, useState } from 'react';
import { useRouteMatch, } from 'react-router-dom';
import Repository from './repository';
import { RepositoryProps, Issue } from './models/repositoryProps';
import api from '../../services/api';

type RepositoryParams = {
    repository: string;
};

const Repository_container: FC = () => {
    const [repositoryy, setRepository] = useState<RepositoryProps | null>(null);
    const [issues, setIssues] = useState<Issue[]>([]);

    const { params } = useRouteMatch<RepositoryParams>();

    useEffect(() => {
        api.get(`repos/${params.repository}`).then(response => {
            setRepository(response.data);
        });

        api.get(`repos/${params.repository}/issues`).then(response => {
            setIssues(response.data);
        });
    }, [params.repository]);


    return (
        <>
            <Repository
                repository={repositoryy}
                issues={issues}
            />
        </>
    );
};

export default Repository_container;