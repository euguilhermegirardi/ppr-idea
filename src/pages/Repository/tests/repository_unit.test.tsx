import { MemoryRouter } from 'react-router-dom';
import { render, RenderResult, screen } from '@testing-library/react';
import Repository from '../repository';
import { mockRepository, mockIssues } from './models/repositoryProps';

describe('Repos', () => {
    let element: RenderResult;

    beforeEach(() => {
        // eslint-disable-next-line testing-library/no-render-in-setup
        element = render(
            <MemoryRouter>
                <Repository
                    repository={mockRepository}
                    issues={mockIssues}
                />
            </MemoryRouter>
        );
    });

    it('should render the component', () => {
        // Arrange
        expect(screen).not.toBeNull();
    });

    it('should find "back" text', () => {
        // Assert
        const backTxt = screen.getByText('Back');

        // Arrange
        expect(backTxt).toBeInTheDocument();
    });

    it('should show image', () => {
        // Assert
        const image = screen.getByTestId('image') as HTMLImageElement;

        // Arrange
        expect(image.src).toContain(mockRepository.owner.avatar_url)
    });

    it('should show main title', () => {
        // Assert
        const mainTitle = screen.getByTestId('full-name');

        // Arrange
        expect(mainTitle).toHaveTextContent(mockRepository.full_name);
    });

    it('should show stars, forks and open issues', () => {
        // Assert
        const stars = screen.getByTestId('stars');
        const forks = screen.getByTestId('forks');
        const issues = screen.getByTestId('issues');

        // Arrange
        expect(stars).toHaveTextContent(mockRepository.stargazers_count);
        expect(forks).toHaveTextContent(mockRepository.forks_count);
        expect(issues).toHaveTextContent(mockRepository.open_issues_count);
    });

    it('should show issues list', () => {
        const issueList = screen.getByTestId('issuesList');

        // eslint-disable-next-line testing-library/no-node-access
        expect(issueList.children).toHaveLength(3);
    });

    it("should match snapshopt", () => {
        // Assert
        const { container } = element;

        // Arrange
        expect(container).toMatchSnapshot();
    });
});