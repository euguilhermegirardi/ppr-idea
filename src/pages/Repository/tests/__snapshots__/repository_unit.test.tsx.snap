// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos should match snapshopt 1`] = `
<div>
  <header
    class="sc-bczRLJ cbCuJc"
  >
    <a
      href="/"
    >
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="15 18 9 12 15 6"
        />
      </svg>
      Back
    </a>
  </header>
  <section
    class="sc-gsnTZi gsjWsv"
  >
    <header>
      <img
        alt="euguilhermegirardi"
        data-testid="image"
        src="https://avatars.githubusercontent.com/u/48716406?v=4"
      />
      <div>
        <strong
          data-testid="full-name"
        >
          euguilhermegirardi/Aircnc-Backend
        </strong>
        <p />
      </div>
    </header>
    <ul>
      <li
        data-testid="stars"
      >
        <strong>
          1
        </strong>
        <span>
          Stars
        </span>
      </li>
      <li
        data-testid="forks"
      >
        <strong>
          0
        </strong>
        <span>
          Forks
        </span>
      </li>
      <li
        data-testid="issues"
      >
        <strong>
          0
        </strong>
        <span>
          Issues
        </span>
      </li>
    </ul>
  </section>
  <div
    class="sc-dkzDqf cQhIhy"
    data-testid="issuesList"
  >
    <a
      data-testid="anchor-tag"
      href="https://github.com/euguilhermegirardi/Aircnc-Frontend/pull/21"
    >
      <div>
        <strong>
          Bump eventsource from 1.0.7 to 1.1.1
        </strong>
        <p>
          dependabot[bot]
        </p>
      </div>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </a>
    <a
      data-testid="anchor-tag"
      href="https://github.com/euguilhermegirardi/Aircnc-Frontend/pull/20"
    >
      <div>
        <strong>
          Bump async from 2.6.3 to 2.6.4
        </strong>
        <p>
          dependabot[bot]
        </p>
      </div>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </a>
    <a
      data-testid="anchor-tag"
      href="https://github.com/euguilhermegirardi/Aircnc-Frontend/pull/19"
    >
      <div>
        <strong>
          Bump minimist from 1.2.5 to 1.2.6
        </strong>
        <p>
          dependabot[bot]
        </p>
      </div>
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </a>
  </div>
</div>
`;
